java-squid/src/main/java/org/sonar/java/ast/visitors/VisitorContext.java

```diff
@@ -19,49 +19,19 @@
  */
 package org.sonar.java.ast.visitors;
 
-import org.sonar.squidbridge.api.SourceCode;
-import org.sonar.squidbridge.api.SourceFile;
-import org.sonar.squidbridge.api.SourceProject;
-
 import java.io.File;
-import java.util.Deque;
-import java.util.LinkedList;
 
 public class VisitorContext {
 
-  private final Deque<SourceCode> sourceCodeStack = new LinkedList<>();
-  private final SourceProject project;
   private File file;
 
-  public VisitorContext(SourceProject project) {
-    if (project == null) {
-      throw new IllegalArgumentException("project cannot be null.");
-    }
-    this.project = project;
-    sourceCodeStack.push(project);
-  }
-
-  private void addSourceCode(SourceCode child) {
-    peekSourceCode().addChild(child);
-    sourceCodeStack.push(child);
-  }
-
-  public SourceCode peekSourceCode() {
-    return sourceCodeStack.peek();
+  public VisitorContext() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1186)